sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftTeamsUserIdentifier.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.common;
+
+import com.azure.android.core.util.CoreUtil;
+
+/**
+ * Communication identifier for Microsoft Teams User
+ */
+public class MicrosoftTeamsUserIdentifier extends CommunicationIdentifier {
+
+    private final String userId;
+    private final boolean isAnonymous;
+    private String id;
+    private CommunicationCloudEnvironment cloudEnvironment = CommunicationCloudEnvironment.PUBLIC;
+
+    /**
+     * Creates a MicrosoftTeamsUserIdentifier object
+     *
+     * @param userId the string identifier representing the identity
+     * @param isAnonymous set this to true if the user is anonymous,
+     *                    for example when joining a meeting with a share link
+     * @throws IllegalArgumentException thrown if id parameter fail the validation.
+     */
+    public MicrosoftTeamsUserIdentifier(String userId, boolean isAnonymous) {
+        if (CoreUtil.isNullOrEmpty(userId)) {
+            throw new IllegalArgumentException("The initialization parameter [userId] cannot be null or empty.");
+        }
+        this.userId = userId;
+        this.isAnonymous = isAnonymous;
+    }
+
+    /**
+     * Creates a MicrosoftTeamsUserIdentifier object
+     *
+     * @param userId the string identifier representing the identity
+     * @throws IllegalArgumentException thrown if id parameter fail the validation.
+     */
+    public MicrosoftTeamsUserIdentifier(String userId) {
+        this(userId, false);
+    }
+
+    /**
+     * @return the string identifier representing the MicrosoftTeamsUserIdentifier object
```

Comment:
Please copy over the comment from C#, this needs to explain that it's the AAD user object id for normal users.

```suggestion
     * @return the id of the Microsoft Teams user. If the user isn't anonymous, the id is the AAD object id of the user.
```